libraries/AP_BattMonitor/AP_BattMonitor_AD7091R5.cpp

```diff
@@ -0,0 +1,236 @@
+#include "AP_BattMonitor_AD7091R5.h"
+
+/**
+ * @brief You can use it to Read Current and voltage of 1-3 batteries from a ADC extender IC over I2C.
+ * AD7091R5 is a ADC extender and we are using it to read current and voltage of multiple batteries.
+ * Examples of Pin combination:
+ *  1)Pin 50 = Voltage 51,52,53 =  Current. For 3 battery combination Voltage will be same accross.
+ *  2)Pin 50,51 = Voltage and Current Battery 1 -   Pin 52,53 = Voltage and Current Battery 2
+ * Only the First instance of Battery Monitor will be reading the values from IC over I2C.
+ * Make sure you understand the method of calculation used in this driver before using it.
+ * e.g. using pin 1 on IC to read voltage of 2 batteries and pin 2 and 3 to read current from individual battery.
+ * Pin number represents 50 = pin 1, 51 = pin 2 and so on 52, 53
+ * BATT2_Monitor  = 24 ,  BATT3_Monitor  = 24
+ * BATT2_VOLT_PIN = 50 ,  BATT3_VOLT_PIN = 50
+ * BATT2_CURR_PIN = 51 ,  BATT3_CURR_PIN = 52
+ *
+ *
+ */
+
+#if AP_BATTERY_AD7091R5_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>
+
+//macro defines
+#define AD7091R5_I2C_ADDR        0x2F // A0 and A1 tied to GND
+#define AD7091R5_I2C_BUS         0
+#define AD7091R5_RESET           0x02
+#define AD7091R5_RESULT_ADDR     0x00
+#define AD7091R5_CHAN_ADDR       0x01
+#define AD7091R5_CONF_ADDR       0x02
+#define AD7091R5_CH_ID(x)        ((x >> 5) & 0x03)
+#define AD7091R5_RES_MASK        0x0F
+#define AD7091R5_REF             3.3f
+#define AD7091R5_RESOLUTION      (float)4096
+#define AD7091R5_PERIOD_USEC     100000
+#define AD7091R5_BASE_PIN        50
+
+
+extern const AP_HAL::HAL& hal;
+const AP_Param::GroupInfo AP_BattMonitor_AD7091R5::var_info[] = {
+
+    // @Param: VOLT_PIN
+    // @DisplayName: Battery Voltage sensing pin on the AD7091R5 Ic
+    // @Description: Sets the analog input pin that should be used for voltage monitoring on AD7091R5.
+    // @Values: -1:Disabled
+    // @User: Standard
+    // @RebootRequired: True
+    AP_GROUPINFO("VOLT_PIN", 30, AP_BattMonitor_AD7091R5, _volt_pin, 0),
```

Comment:
30 to 39 are reserved for the DroneCAN monitor. Looks like you would have to start at 56.